index.js

```diff
@@ -320,12 +337,10 @@
   //  xor :: a -> a -> a
   S.xor = R.curry(function(x, y) {
     assertTypeMatch(x, y);
-    switch (true) {
-      case toBoolean(x) !== toBoolean(y): return or(x, y);
-      case R.is(Array, x):                return [];
-      case R.is(Boolean, x):              return false;
-      default:                            return x.empty();
-    }
+    var xBool = toBoolean(x);
+    var yBool = toBoolean(y);
+    var xEmpty = empty(x);
+    return xBool !== yBool ? or(x, y) : xEmpty;
```

Comment:
> It may be worth leaving a comment here explaining that empty() has the intended side effecting of throwing a `TypeError`.

We certainly could, though making the change from my previous comments would cause tests to fail, so a comment isn't necessary to prevent regression.

> Also, what's special about the empty value of the first element?

Nothing. `empty(x)` and `empty(y)` should be equivalent, since `x` and `y` are values of the same type (as asserted by `assertTypeMatch(x, y)`).